src/xma/include/lib/xma_buffer.hpp

```diff
@@ -0,0 +1,62 @@
+/**
+ * Copyright (C) 2019 Xilinx, Inc
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"). You may
+ * not use this file except in compliance with the License. A copy of the
+ * License is located at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+#ifndef _XMA_BUFFERCLS_H_
+#define _XMA_BUFFERCLS_H_
+
+#include <cstdint>
+#include <cstring>
+#include <string>
+#include <atomic>
+#include "xrt.h"
+
+namespace xma_core {
+namespace plg {
+
+/**
+ * class buffer : abstraction for xma buffer
+ */
+class buffer
+{
+private:
```

Comment:
What is the point of this class in light of the XRT native APIs?   I would hope that whatever you build in XMA is not duplicating what the native APIs are providing?  Consider using xrt::bo.

Reply:
Changed to derived class from xrt::bo